tags/terminal/readline_vi.py

```diff
@@ -0,0 +1,187 @@
+from dataclasses import dataclass
+from typing import Callable
+
+from talon import Context, actions
+
+sleep = actions.sleep
+
+ctx = Context()
+ctx.matches = """
+tag: user.readline_vi
+"""
+
+
+@dataclass
+class PendingSelection:
+    motion: str
+    endAction: str
+    count: int = 1
+
+
+pendingSelection: PendingSelection | None = None
+
+
+# using d instead of c for removal so that we remain in normal mode and the character sent to exit normal mode doesn't get inserted
+edit_action_vi_keys: dict[str, str] = {
+    "cut": "d",
+    "delete": "d",
+    "goBefore": "",
+    "goAfter": "",
+    # We can't actually use the clipboard for these operations
+    # "copyToClipboard": "y",
+    # "cutToClipboard": "d",
+    # "pasteFromClipboard": "y",
+}
+
+
+def normal_cmd(keys):
+    actions.key("escape")
+    # sleep to avoid interpreting as an escape sequence
+    sleep(0.2)
+    actions.insert(keys)
```

Comment:
Given that you have several places where you need to duplicate this logic, consider using `actions.key` here instead.

Reply:
I am not at all clear what you are asking,  or what the benefits of actions . key is, Could you please clarify?